tests/run-pass/crate_traversal.rs

```diff
@@ -1,16 +1,164 @@
-// A small trait to test crate traversal
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+//
 
-trait TestTrait {
-    fn test_method() {}
+// statements and expressions that exercise all visitor code paths
+
+#![allow(const_err)]
+#![allow(unused)]
+#![feature(asm)]
+#![feature(box_syntax)]
+#![feature(generators)]
+
+use std::mem;
+
+pub trait TestTrait {
+    fn test_method() {
+        let _loc = [1];
+        let loc2 = [ 0; 512 ];
+    }
 }
 
-struct TestStruct {
-     test_field: i64
+pub struct TestStruct {
+    pub test_field: i64
 }
 
 impl TestStruct {
-    fn test_function() {}
-    fn test_method(&self) {}
+    pub fn test_function() {}
+    pub fn test_method(&self) {}
+}
+
+pub unsafe fn test1() {
```

Comment:
Some test method names and/r comments describing which features each test is targeting might be useful? I can figure some of them out, but not all :).

Reply:
In general, yes, such comments are very helpful (and I wish the rust compiler's test suite had more of them). This test file, however, is of a temporary nature and will become redundant as more useful tests with expected behavior and so on come along. See issue #10. Adding comments detailing with MIR features each test uses would be very tedious and is not a top priority for me right now.